### ISSUE TITLE
dispatch 0.3.0 needs ppx and therefore depends on ocaml>=4.02

### ISSUE THREAD

**Issue by ansiwen** [inhabitedtype/ocaml-dispatch#15]
Installing dispatch 0.3.0 with opam and ocaml 4.01 fails, because it needs ppx, which is not available.

**Comment by seliopou** [comment]
Thanks for opening the issue! A few questions:
- Did you have js_of_ocaml installed when trying to install dispatch?
- If so, which version?

**Comment by ansiwen** [comment]
Here more infos:

Mac OS X 10.11, El Capitan

```
$ ocaml -version
The OCaml toplevel, version 4.01.0
```

```
$ opam --version
1.2.2
```

```
$ opam install dispatch
The following actions will be performed:
  ∗  install dispatch 0.3.0

=-=- Gathering sources =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=  🐫 
[dispatch] Archive in cache

=-=- Processing actions -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=  🐫 
[ERROR] The compilation of dispatch failed at "ocaml setup.ml -configure --prefix
        /Users/sanders/.opam/4.01.0 --enable-js-of-ocaml".
Processing  1/1: [dispatch: ocamlfind remove]
#=== ERROR while installing dispatch.0.3.0 ====================================#
# opam-version 1.2.2
# os           darwin
# command      ocaml setup.ml -configure --prefix /Users/sanders/.opam/4.01.0 --enable-js-of-ocaml
# path         /Users/sanders/.opam/4.01.0/build/dispatch.0.3.0
# compiler     4.01.0
# exit-code    1
# env-file     /Users/sanders/.opam/4.01.0/build/dispatch.0.3.0/dispatch-87638-1872ec.env
# stdout-file  /Users/sanders/.opam/4.01.0/build/dispatch.0.3.0/dispatch-87638-1872ec.out
# stderr-file  /Users/sanders/.opam/4.01.0/build/dispatch.0.3.0/dispatch-87638-1872ec.err
### stderr ###
# ocamlfind: Package `js_of_ocaml.ppx' not found
# W: Field 'pkg_js_of_ocaml_ppx' is not set: Command ''/Users/sanders/.opam/4.01.0/bin/ocamlfind' query -format %d js_of_ocaml.ppx > '/var/folders/r1/mgrztsmx2ml04nsfbvr427mr0000gn/T/oasis-cd6700.txt'' terminated with error code 2
# E: Cannot find findlib package js_of_ocaml.ppx
# E: Failure("1 configuration error")



=-=- Error report -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=  🐫 
The following actions failed
  ∗  install dispatch 0.3.0
No changes have been performed
```

```
$ opam show js_of_ocaml 
             package: js_of_ocaml
             version: 2.7
          repository: default
        upstream-url: https://github.com/ocsigen/js_of_ocaml/archive/2.7.tar.gz
       upstream-kind: http
   upstream-checksum: 99974e47a97a5b2fd2bbd0c58759cec1
            homepage: http://ocsigen.org/js_of_ocaml
         bug-reports: https://github.com/ocsigen/js_of_ocaml/issues
            dev-repo: https://github.com/ocsigen/js_of_ocaml.git
              author: Ocsigen team
             depends: cmdliner & base-unix & ocamlfind >= 1.5.1 & lwt >= 2.4.4 & menhir & cppo & camlp4 & base64 >= 2.0.0 & (base-no-ppx | ppx_tools) & ocamlbuild
             depopts: deriving | ppx_deriving | tyxml | reactiveData
   installed-version: 2.7 [4.01.0]
  available-versions: 1.2, 1.3.1, 1.3.2, 1.4.0, 2.0, 2.1, 2.2, 2.3, 2.4, 2.4.1, 2.5, 2.6, 2.7
         description: Compiler from OCaml bytecode to Javascript
```
